### ISSUE TITLE
Text and Binary outputs on writers

### ISSUE THREAD

**Issue by navispatial** [navibyte/geospatial#126]
`BaseWriter` before changes:

```dart
/// An interface to write objects into some content format.
// ignore: one_member_abstracts
abstract class BaseWriter {
  /// A string representation of content already written to this (text) writer.
  ///
  /// Must return a valid string representation when this writer is writing to
  /// a text output. If an output does not support a string representation then
  /// returned representation is undefined.
  @override
  String toString();
}
```

Change the contract of this interface to support also binary outputs.

```dart
/// An interface to write objects into some content format (text or binary).
abstract class BaseWriter {
  /// The text representation of content already written to this writer.
  ///
  /// Returns a valid string representation of content for a text format writer
  /// and a base64 string representation of content for a binary format writer.
  @override
  String toString();

  /// The binary representation of content already written to this writer.
  ///
  /// Returns a valid binary representation of content as a sequence of bytes
  /// for a binary format writer, and an UTF8 encoded binary representation of
  /// textual content for a text format writer.
  Uint8List toBytes();
```

**Comment by navispatial** [comment]
New definition:

```dart
import 'dart:typed_data';

/// An interface to write content of [T] into a content format (text or binary).
abstract class ContentWriter<T extends Object> {
  /// The content [output] interface that is used by a client to write content.
  T get output;

  /// The text representation of content already written to this writer.
  ///
  /// Returns a valid string representation of content for a text format writer
  /// and a base64 string representation of content for a binary format writer.
  @override
  String toString();

  /// The binary representation of content already written to this writer.
  ///
  /// Returns a valid binary representation of content as a sequence of bytes
  /// for a binary format writer, and an UTF8 encoded binary representation of
  /// textual content for a text format writer.
  Uint8List toBytes();
}
```

**Comment by navispatial** [comment]
Some changes still planned....

```dart
/// An interface to encode some content of [T] into a text or binary format.
abstract class ContentEncoder<T extends Object> {
  /// The [content] interface that is used by a client to encode content.
  ///
  /// Methods of [T] (that are used to write content when encoding) should throw
  /// `FormatException` if encoding fails.
  T get content;

  /// The text representation of content already written to this encoder.
  ///
  /// Returns a valid string representation of content for a text format encoder
  /// and a base64 string representation of content for a binary format encoder.
  ///
  /// Throws `FormatException` if encoding fails.
  @override
  String toString();

  /// The binary representation of content already written to this encoder.
  ///
  /// Returns a valid binary representation of content as a sequence of bytes
  /// for a binary format encoder, and an UTF8 encoded binary representation of
  /// textual content for a text format encoder.
  ///
  /// Throws `FormatException` if encoding fails.
  Uint8List toBytes();
}
```

**Comment by navispatial** [comment]
Implemented in geobase [0.3.0-dev.1](https://pub.dev/packages/geobase/versions/0.3.0-dev.1) as:

```dart
import 'dart:typed_data';

/// An interface to encode [Content] into a text or binary format.
abstract class ContentEncoder<Content extends Object> {
  /// Returns the [Content] writer.
  ///
  /// Calling this property never throws, but methods provided by the [Content]
  /// interface (that are used to write content) should throw `FormatException`
  /// if writing / encoding fails.
  Content get writer;

  /// The binary representation of content already written to this encoder.
  ///
  /// Returns a valid binary representation of content as a sequence of bytes
  /// for a binary format encoder, and an UTF8 encoded binary representation of
  /// textual content for a text format encoder.
  ///
  /// Throws `FormatException` if encoding fails.
  Uint8List toBytes();

  /// The text representation of content already written to this encoder.
  ///
  /// Returns a valid string representation of content for a text format encoder
  /// and a base64 string representation of content for a binary format encoder.
  ///
  /// Throws `FormatException` if encoding fails.
  String toText();

  /// The text representation of content already written to this encoder.
  ///
  /// Equals to calling [toText].
  ///
  /// Returns a valid string representation of content for a text format encoder
  /// and a base64 string representation of content for a binary format encoder.
  ///
  /// Throws `FormatException` if encoding fails.
  @override
  String toString();
}
```
